src/vs/workbench/contrib/terminalContrib/find/browser/terminal.find.contribution.ts

```diff
@@ -3,65 +3,90 @@
-	if (!result) {
-		const terminalService = accessor.get(ITerminalService);
-		const widget = accessor.get(IInstantiationService).createInstance(TerminalFindWidget, instance);
-
-		// Track focus and set state so we can force the scroll bar to be visible
-		let focusState = false;
-		widget.focusTracker.onDidFocus(() => {
-			focusState = true;
-			instance.forceScrollbarVisibility();
-			terminalService.setActiveInstance(instance);
-		});
-		widget.focusTracker.onDidBlur(() => {
-			focusState = false;
-			instance.resetScrollbarVisibility();
-		});
 
-		// Attach the find widget and listen for layout
-		instance.registerChildElement({
-			element: widget.getDomNode(),
-			layout: dimension => widget.layout(dimension.width),
-			xtermReady: xterm => {
-				xterm.onDidChangeFindResults(() => widget.updateResultCount());
-			}
-		});
+	private _findWidget: Lazy<TerminalFindWidget>;
+	private _lastLayoutDimensions: IDimension | undefined;
+	private _focusState: boolean = false;
+
+	get findWidget(): TerminalFindWidget { return this._findWidget.value; }
 
-		// Cache the widget while the instance exists, dispose it when the terminal is disposed
-		instance.onDisposed(e => {
-			const focusTerminal = focusState;
-			widget?.dispose();
-			findWidgets.delete(e);
-			if (focusTerminal) {
-				instance.focus();
+	constructor(
+		private readonly _instance: ITerminalInstance,
+		processManager: ITerminalProcessManager,
+		widgetManager: TerminalWidgetManager,
+		@IInstantiationService readonly _instantiationService: IInstantiationService,
+		@ITerminalService readonly _terminalService: ITerminalService
+	) {
+		super();
+
+		this._findWidget = new Lazy(() => {
+			const findWidget = this._instantiationService.createInstance(TerminalFindWidget, _instance);
+
+			// Track focus and set state so we can force the scroll bar to be visible
+			findWidget.focusTracker.onDidFocus(() => {
+				this._focusState = true;
+				_instance.forceScrollbarVisibility();
+				_terminalService.setActiveInstance(_instance);
+			});
+			findWidget.focusTracker.onDidBlur(() => {
+				this._focusState = false;
+				_instance.resetScrollbarVisibility();
+			});
+
+			_instance.domElement.appendChild(findWidget.getDomNode());
+			if (this._lastLayoutDimensions) {
+				findWidget.layout(this._lastLayoutDimensions.width);
 			}
+
+			return findWidget;
 		});
+	}
+
+	layout(xterm: IXtermTerminal & { raw: RawXtermTerminal }, dimension: IDimension): void {
+		this._lastLayoutDimensions = dimension;
+		this._findWidget.rawValue?.layout(dimension.width);
+	}
+
+	xtermReady(xterm: IXtermTerminal & { raw: RawXtermTerminal }): void {
+		this._register(xterm.onDidChangeFindResults(() => this._findWidget.rawValue?.updateResultCount()));
+	}
+
+	override dispose() {
+		super.dispose();
 
-		findWidgets.set(instance, widget);
-		result = widget;
+		const focusTerminal = this._focusState;
+		this._findWidget.rawValue?.dispose();
+		if (focusTerminal) {
+			this._instance.focus();
```

Comment:
Is this focus on dispose needed :thinking: ? it was in the original code

Reply:
I guess not. I want terminalContribs to work similar to xterm.js addons where they can be enabled/disabled without the terminal disposing, but still it should be up to the terminal instance to restore focus in this case so I think we can safely remove this